src/test/regress/expected/pg15.out

```diff
@@ -552,7 +552,36 @@ SELECT count(*)=100 FROM copy_test2;
  t
 (1 row)
 
+-- test NULL NOT DISTINCT clauses
+-- set the next shard id so that the error messages are easier to maintain
+SET citus.next_shard_id TO 960050;
+CREATE TABLE null_distinct_test(id INT, c1 INT, c2 INT, c3 VARCHAR(10)) ;
+SELECT create_distributed_table('null_distinct_test', 'id');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE UNIQUE INDEX idx1_null_distinct_test ON null_distinct_test(id, c1) NULLS DISTINCT ;
+CREATE UNIQUE INDEX idx2_null_distinct_test ON null_distinct_test(id, c2) NULLS NOT DISTINCT ;
+INSERT INTO null_distinct_test VALUES (1, 1, 1, 'data1') ;
+INSERT INTO null_distinct_test VALUES (1, 2, NULL, 'data2') ;
+INSERT INTO null_distinct_test VALUES (1, NULL, 3, 'data3') ;
+-- should fail as we already have a null value in c2 column
+INSERT INTO null_distinct_test VALUES (1, NULL, NULL, 'data4') ;
```

Comment:
lets play with ON CONFLICT as well:
```SQL
-- on conflict DO NOTHING is a no-op
INSERT INTO null_distinct_test VALUES (1, NULL, NULL, 'data4')  ON CONFLICT DO NOTHING;
```

and, maybe if possible one more test with ON CONFLICT .. TO UPDATE  ..